dali/kernels/common/find/find_first_last_gpu_test.cu

```diff
@@ -0,0 +1,211 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/kernels/common/find/find_first_last_gpu.cuh"
+#include <gtest/gtest.h>
+#include <vector>
+#include "dali/core/cuda_event.h"
+#include "dali/kernels/dynamic_scratchpad.h"
+#include "dali/pipeline/data/views.h"
+#include "dali/test/tensor_test_utils.h"
+#include "dali/test/test_tensors.h"
+
+namespace dali {
+namespace kernels {
+namespace find {
+namespace test {
+
+template <typename T>
+struct threshold {
+  T value_;
+  explicit threshold(T value = 0) : value_(value) {}
+
+  DALI_HOST_DEV DALI_FORCEINLINE bool operator()(T x) const noexcept {
+    return x >= value_;
+  }
+};
+
+template <typename T>
+void SequentialFillBothEnds(T* data, int64_t data_len, T start_value) {
+  auto value = start_value;
+  int i = 0;
+  int j = data_len - 1;
+  for (; i < j; value++) {
+    data[i++] = value;
+    data[j--] = value;
+  }
+  if (i == j)
+    data[i] = value;
+}
+
+class FindFirstLastTestGPU : public ::testing::Test {
+ public:
+  using T = float;
+  TestTensorList<T> in_;
+  TestTensorList<int64_t> out_begin_;
+  TestTensorList<int64_t> out_length_;
+
+  TestTensorList<int64_t> ref_begin_;
+  TestTensorList<int64_t> ref_length_;
+
+  using Predicate = threshold<T>;
+  Predicate thresh{3};
+  span<Predicate> predicates{&thresh, 1};
+
+  void SetUp() final {
+    int nsamples = 5;
+
+    TensorListShape<> sh = {{5, }, {10, }, {9, }, {7, }, {1000, }};
```

Comment:
I would like to see a test that covers more than one block to reduce (longer than 32*32).